test/e2e-test/suite_test.go

```diff
@@ -100,6 +107,30 @@ var _ = BeforeSuite(func(done Done) {
 	// For some reason, traitDefinition is created as a Cluster scope object
 	Expect(k8sClient.Create(context.Background(), &manualscalertrait)).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
 	By("Created manual scalar trait definition")
+
+	// Create manual scaler trait definition with spec.extension field
+	definitionExtension := DefinitionExtension{
+		Alias: "ManualScaler",
+	}
+	in := new(runtime.RawExtension)
+	in.Raw, _ = json.Marshal(definitionExtension)
+
+	extendedmanualscalertrait = v1alpha2.TraitDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "manualscalertraits-extended.core.oam.dev",
+			Labels: map[string]string{"trait": "manualscalertrait"},
+		},
+		Spec: v1alpha2.TraitDefinitionSpec{
+			WorkloadRefPath: "spec.workloadRef",
+			Reference: v1alpha2.DefinitionReference{
+				Name: "manualscalertraits-extended.core.oam.dev",
+			},
+			Extension: in,
+		},
+	}
+	Expect(k8sClient.Create(context.Background(), &extendedmanualscalertrait)).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
```

Comment:
I didn't see get only create here?

Reply:
It did.
- Before e2e-test
```
✗ k get traitdefinition
NAME                             DEFINITION-NAME
manualscalertrait.core.oam.dev   manualscalertrait.core.oam.dev
```

- During e2e-test
```
➜  /Users/zhouzhengxi k get traitdefinition --watch
NAME                                       DEFINITION-NAME
foos.example.com                           foos.example.com
manualscalertrait.core.oam.dev             manualscalertrait.core.oam.dev
manualscalertraits-extended.core.oam.dev   manualscalertraits-extended.core.oam.dev
manualscalertraits.core.oam.dev            manualscalertraits.core.oam.dev
foos.example.com                           foos.example.com
manualscalertraits.core.oam.dev            manualscalertraits.core.oam.dev
manualscalertraits-extended.core.oam.dev   manualscalertraits-extended.core.oam.dev
^C%
```

- After e2e-test
```                                                                                                                                                                                                         ➜  /Users/zhouzhengxi k get traitdefinition
NAME                             DEFINITION-NAME
manualscalertrait.core.oam.dev   manualscalertrait.core.oam.dev
```
manualscalertraits-extended.core.oam.dev with spec.extension was created and destroyed.